lib/modules/apostrophe-palette-widgets/public/js/leanUser.js

```diff
@@ -33,109 +33,118 @@ if (window.$) {
-    var unit = $field.attr('data-apos-palette-unit') || "";
-    var valueTemplate = $field.attr('data-apos-value-template') || false;
-    var value = $field.find(':input:not([data-apos-workflow-field-state-control])').val();
+  apos.paletteWidgets.debouncedSaves = {};
+}
 
-    if (!value) {
+function init($form, $tag, schema, piece) {
+  return apos.schemas.populate($form, schema, piece, function (err) {
+    if (err) {
+      apos.notify('A problem populating/enhancing the palette field', {
+        type: 'error',
+        dismiss: true
+      });
       return;
     }
 
-    if (selector && property) {
-      property = property.split(',');
-      _.each(property, function (prop) {
-        var rule = "";
-        if (mediaQuery) {
-          rule = '@media ' + mediaQuery + ' { ';
-        }
-
-        if (valueTemplate) {
-          var regex = /%VALUE%/gi;
-          rule = rule + selector + " { " + prop + ": " + valueTemplate.replace(regex, value + unit) +  "; } ";
-        } else {
-          rule = rule + selector + " { " + prop + ": " + value + unit + "; } ";
-        }
-
-        if (mediaQuery) {
-          rule = rule + ' }';
-        }
-        $tag.append(rule);
-      });
-    }
-  };
+    listen($form, $tag, schema, piece);
 
-  function save ($field, fieldValue, fieldSchema, $tag) {
-    // build an object of data from the palette field to update the apostrophe-global piece with
-    var save = {};
-    save.field = {
-      name: fieldSchema.name,
-      value: $field.find(':input:not([data-apos-workflow-field-state-control])').val()
-    };
-    save._id = apos.modules['apostrophe-palette-widgets'].options.piece._id;
+  });
+};
 
-    return send(function(err) {
-      if (err) {
-        apos.notify('Something was not right. Please review your submission.', { type: 'error', dismiss: true });
+// Listen for changes to form elements within the palette tool
+function listen($form, $tag, schema, piece) {
+  $form.find(':input:not([data-apos-workflow-field-state-control])').change(function () {
+    var $field = $(this).parents('[data-apos-palette-field]');
+    var fieldValue = $field.find(':input:not([data-apos-workflow-field-state-control])').val();
+    var fieldSchema = _.find(schema, function (field) {
+      if ($field.attr('data-name') === field.name) {
+        return field;
       }
     });
+    submit($field, fieldValue, fieldSchema, $tag);
+  });
+};
+
+// Handles prepping the current front end for changes as well as sending
+// updates to the apostrophe-global piece
+function submit ($field, fieldValue, fieldSchema, $tag) {
+  apply($field, fieldValue, fieldSchema, $tag);
+  var saveField = apos.paletteWidgets.debouncedSaves[fieldSchema.name];
+  if (!saveField) {
+    saveField = _.debounce(save, 250, {
+      leading: true,
+      trailing: true
+    });
+    apos.paletteWidgets.debouncedSaves[fieldSchema.name] = saveField;
+  }
+  saveField($field, fieldValue, fieldSchema, $tag);
```

Comment:
Hard to tell with all the moving, but this used to be `save(...)`. `saveField` wasn't actually being used.

Reply:
see https://github.com/apostrophecms/apostrophe-palette/pull/20/commits/562570c88d4180b6e12e456e7f019b6f0e25b0f8

Reply:
This does sound like an improvement, has it been tested?

Reply:
Yep, I tested it on the Scottie project, which is lean and uses palette.